app/facades/food_facade.rb

```diff
@@ -0,0 +1,10 @@
+class FoodFacade
+  def self.all_foods(food)
+    json_data = FoodService.foods(food)
+    food_hits = json_data[:totalHits]
+
+    foods = json_data[:foods].map do |food|
+      Food.new(food, food_hits)
+    end
```

Comment:
In relation to the comment below on your poro, if instead of saving the total hits to each `Food` instance, you could return the hits and foods in an array at the end of this method, like this:

```ruby
  def self.all_foods(food)
    json_data = FoodService.foods(food)
    food_hits = json_data[:totalHits]

    foods = json_data[:foods].map do |food|
      Food.new(food, food_hits)
    end
    [food_hits, foods]
  end
end
```

And then in your controller do something like this:
```ruby
...code
  else
     @food_hits, @foods = FoodFacade.all_foods(@food)
  end
```